zio-http/src/main/scala/zhttp/http/Method.scala

```diff
@@ -18,7 +18,7 @@ object Method {
   object CONNECT extends Method
 
   def fromJHttpMethod(method: JHttpMethod): Method =
-    method match {
+    (method: @unchecked) match {
```

Comment:
Should it be an `Option` ?

Reply:
I added this because it was complaining about not being an exhaustive match under scala 3

Reply:
I guess the API should be 
```scala
def fromJHttpMethod(method: JHttpMethod): Option[Method]
```

Option wrapping has a minor overhead in terms of performance that's we ignored it.

Reply:
Yeah that makes perfect sense.